library/modules/Items.cpp

```diff
@@ -2089,7 +2090,7 @@ int Items::getValue(df::item *item, df::caravan_state *caravan)
     return value;
 }
 
-int32_t Items::createItem(df::item_type item_type, int16_t item_subtype, int16_t mat_type, int32_t mat_index, df::unit* unit) {
+bool Items::createItem(std::vector<df::item *> &out_items, df::unit* unit, df::item_type item_type, int16_t item_subtype, int16_t mat_type, int32_t mat_index, int32_t growth_print) {
```

Comment:
can the `createitem` plugin call this API as well so we don't duplicate code?

Reply:
The `createitem` plugin has separate options for placing the items into containers/buildings, while this function always places them on the floor. Unless we want this function to provide *all* of the functionality of the plugin, they should stay separate.

Reply:
I think the placement could be kept separate, but can the core creation logic be factored out?

Reply:
Done.